### PR TITLE
import: remove the proper path

### LINKED ISSUE #70: caching issue in stacker build
1. import a dir with one file in it in a stacker.yaml file
2. do a stacker build
3. touch a file in that dir
4. do a stacker build
5. remove the file in that you additionally created
6. do a stacker build, it reports found a cached layer
7. In fact it found the wrong cached layer, when you unpack you see the file you created in step 3 in the layer

### PR DESCRIPTION
**Pull request by hallyn**
Closes #70

When a file under an imported directory is removed, calculate the
correct path to remove.

Signed-off-by: Serge Hallyn <shallyn@cisco.com>

### CODE REVIEW

File: import.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func importFile(imp string, cacheDir string) (string, error) {
 	for _, d := range diff {
 		switch d.Type() {
 		case mtree.Missing:
-			err := os.RemoveAll(path.Join(cacheDir, d.Path()))
+			err := os.RemoveAll(path.Join(cacheDir, path.Base(imp), d.Path()))
 			if err != nil {
 				return "", errors.Wrapf(err, "couldn't remove missing import %s", path.Join(cacheDir, d.Path()))
```
Comment (tych0) [inline import.go]: Can you fix the path in the error message too? Otherwise lgtm.

**Comment by andaaron** [comment]
Maybe we should add a test which reproduces the conditions in which the bug was found.

**Comment by hallyn** [comment]
> Maybe we should add a test which reproduces the conditions in which the bug was found.

Oh, actually there was a test, but it was wrong.  Trying to fix it.  But man bats, is annoying to debug.

**Comment by hallyn** [comment]
> > Maybe we should add a test which reproduces the conditions in which the bug was found.
> 
> Oh, actually there was a test, but it was wrong. Trying to fix it. But man bats, is annoying to debug.

sorry, yes, i did find it later that night.

**Pull request merged**